### DISCUSSION TITLE
Fastest way to adjust subtitle fps and delay for any scenario

### DISCUSSION THREAD

**Discussion by forthrin** [mpv-player/mpv#14706] [Q&A]
What is the intended workflow of `—sub-fps` and `—sub-delay` options, and how well does this cover all use cases?

If the framerate matches, it's usually *relatively* easy, but if both delay (eg. segment added/removed in the beginning) and framerate factor (may be quite arbitrary at times) are off, you have to fumble a *lot* to get the two values just right.

Save speech recognition, the following steps seem necessary to find the perfect adjustments for any situation:

1. Play the video and write down observed time codes of the first and last spoken sentence.
2. Use a script to calculate an offset and “stretch” value*  and adjust all timecodes accordingly.
3. Play the video again, and repeat step two with slight adjustments until the timing is perfect.

*) `(observed_last - observed_first) / (file_last - file_first)`

Is the current implementation in mpv optimal or even useful? Does this even belong in mpv? What is the best approach?

**Comment by forthrin** [comment]
@kasper93: Great obsoleting stale reports, but reopen this one and move to Discussions?

**Comment by jamesmcm** [comment]
Here is a Python script I generated with Claude to do this quickly: https://gist.github.com/jamesmcm/2869c5c96471f12c09ad87dd76d053f0

There you can just fill in 
```python
        reference_points = [
            ("00:02:36,833", "01:26:42"),
            ("00:03:30,129", "01:27:37"),
            ("01:27:47,028", "02:48:27")
        ]
```

With the SRT timestamp on the left, and the actual video timestamp on the right on line 87.

And then run it like:
```bash
$ ./subs.py input.srt output.srt
```

It's still a pain having to manually enter the reference points (especially skipping near the end of a film!) but it's better than  manually doing it repeatedly.
